File: Frontend/src/app/business-logic/ServiceFactors.ts
import { Port } from "./Port";
export class ServiceFactors {
    public mongoDiscovered:boolean = false;
    public  cassandraDiscovered:boolean = false;
    public  msSQLDiscovered:boolean = false;
    public  mySQLDiscovered:boolean = false;
    public  redisDiscovered:boolean = false;
    public  vncDiscovered:boolean = false;
    public  rdpDiscovered:boolean = false;
    public  rsyncDiscovered:boolean = false;
    public  imapDiscovered:boolean = false;
    public  ftpDiscovered:boolean = false;
    public  smbDiscovered:boolean = false;
    public  telnetDiscovered:boolean = false;
    public  pop3Discovered:boolean = false;
    public serviceDict:Map<String, Boolean> | undefined;
    public ports:Array<Port> | undefined;

}

File: Backend/src/main/java/com/example/riskfactors/model/ServiceFactors.java
package com.example.riskfactors.model;

import lombok.*;

import javax.persistence.*;
import java.util.List;
import java.util.Map;
@Setter
@Getter
@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ServiceFactors {

    @Id
    @SequenceGenerator(
            name = "serviceFactorsSequence",
            sequenceName = "serviceFactorsSequence",
            allocationSize = 1
    )
    @GeneratedValue(
            strategy = GenerationType.SEQUENCE,
            generator = "serviceFactorsSequence"
    )
    private Long ServiceFactorsId;
    private Boolean mongoDiscovered;
    private Boolean cassandraDiscovered;
    private Boolean msSQLDiscovered;
    private Boolean mySQLDiscovered;
    private Boolean redisDiscovered;
    private Boolean vncDiscovered;
    private Boolean rdpDiscovered;
    private Boolean rsyncDiscovered;
    private Boolean imapDiscovered;
    private Boolean ftpDiscovered;
    private Boolean smbDiscovered;
    private Boolean telnetDiscovered;
    private Boolean pop3Discovered;
    @ElementCollection
    private Map<String, Boolean> serviceDict;
    @OneToMany(
            cascade = CascadeType.ALL
    )
    @JoinColumn(
            name = "serviceFactorsFK"
    )
    private List<Port> ports;

}
